resource/js/global-search.js

```diff
@@ -0,0 +1,431 @@
+        showNotation: null
+      }
+    },
+    computed: {
+      vocabSelectorPlaceholder () {
+        return $t('1. Valitse sanasto')
+      },
+      langSelectorPlaceholder () {
+        return $t('2. Valitse kieli')
+      },
+      searchPlaceholder () {
+        return $t('3. Kirjoita hakutermi')
+      },
+      noResults () {
+        return $t('No results')
+      },
+      selectSearchLanguageAriaMessage () {
+        return $t('Select search language')
+      },
+      textInputWithDropdownButtonAriaMessage () {
+        return $t('Text input with dropdown button')
+      },
+      searchAriaMessage () {
+        return $t('Search')
+      },
+      getSelectedVocabs () {
+        return this.selectedVocabs.map(key => ({ key, value: this.vocabStrings[key] }))
+      },
+      selectedVocabsString () {
+        return this.getSelectedVocabs.map(voc => voc.value).join(' ')
+      }
+    },
+    mounted () {
+      this.languages = window.SKOSMOS.languageOrder
+      this.selectedLanguage = this.parseSearchLang()
+      this.languageStrings = window.SKOSMOS.language_strings
+      this.vocabStrings = window.SKOSMOS.vocab_list
+    },
+    watch: {
+      selectedLanguage (newLang) {
+        if (!newLang) return
+        const url = new URL(window.location.href)
+        if (newLang === 'all') {
+          url.searchParams.set('anylang', 'on')
+        } else {
+          url.searchParams.set('clang', newLang)
+          url.searchParams.set('lang', newLang)
+          url.searchParams.delete('anylang')
+        }
+        window.history.replaceState({}, '', url.toString())
+      }
+    },
+    methods: {
+      autoComplete () {
+        const delayMs = 300
+
+        // when new autocomplete is fired, empty the previous result
+        this.renderedResultsList = []
+
+        // cancel the timer for upcoming API call
+        clearTimeout(this._timerId)
+        this.hideAutoComplete()
+
+        // delay call, but don't execute if the search term is not at least two characters
+        if (this.searchTerm.length > 1) {
+          this._timerId = setTimeout(() => { this.search() }, delayMs)
+        }
+      },
+      search () {
+        const mySearchCounter = this.searchCounter + 1 // make sure we can identify this search later in case of several ongoing searches
+        this.searchCounter = mySearchCounter
+        let skosmosSearchUrl = window.SKOSMOS.baseHref + 'rest/v1/search?'
+        const skosmosSearchUrlParams = this.formatSearchUrlParams()
+        skosmosSearchUrl += skosmosSearchUrlParams.toString()
+
+        fetch(skosmosSearchUrl)
+          .then(data => data.json())
+          .then(data => {
+            if (mySearchCounter === this.searchCounter) {
+              this.renderedResultsList = data.results // update results (update cache if it is implemented)
+              this.renderResults() // render after the fetch has finished
+            }
+          })
+      },
+      formatSearchUrlParams () {
```

Comment:
This method is used for two different purposes:
- generating parameters for the search result page URL
- generating parameters for the REST API `search` method

This leads to unfortunate consequences: some search result page URL parameters have been renamed (`q` -> `query`, `vocab` -> `vocabs`, and it is also given unnecessary parameters `lang` and `unique`).

I suggest splitting this into two methods: one for the search result page and another for the REST API.